packages/components/src/utils/validation/validateProps.ts

```diff
@@ -241,3 +248,29 @@ export const validateProps = <T extends Class<any>>(instance: InstanceType<T>, p
       printErrorMessage({ ...error, componentName: getTagNameWithoutPrefix(instance.host as HTMLElement) })
     );
 };
+
+/**
+ * Validates an array using a provided validator function and returns the first encountered validation error.
+ *
+ * @param {string} propName - The name of the property being validated.
+ * @param {any} arr - The input to be validated.
+ * @param {ValidatorFunction} validator - The validator function that checks each array item.
+ * @returns {ValidationError | undefined} The first encountered validation error object, or undefined if the array is valid.
+ */
+export const isValidArray = (propName: string, arr: any, validator: ValidatorFunction): ValidationError => {
+  // TODO: Would be better to return the expected array type in this case as well
+  if (!Array.isArray(arr)) {
+    return {
+      propName,
+      propValue: arr,
+      propType: '[]',
+    };
+  }
+  for (const item of arr) {
+    const validationError = validator(propName, item);
+    if (validationError) {
+      return { ...validationError, propType: `${validationError.propType}[]` };
```

Comment:
Please refactor function to single exit. Perhaps using array functions to get the error would be easier to read and understand than abortion criteria within a loop.